torchtune/models/clip/_transforms.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import Dict, List, Optional, Tuple
+
+import torch
+import torchvision
+from PIL import Image
+
+from torchtune.modules.transforms.vision import (
+    get_canvas_best_fit,
+    resize_with_pad,
+    tile_crop,
+)
+from torchtune.modules.transforms.vision.get_canvas_best_fit import (
+    find_supported_resolutions,
+)
+
+from torchvision.transforms.v2 import functional as F
+
+logger = logging.getLogger(__name__)
+
+
+class CLIPImageTransform:
+    """
+    This class accepts images of any size and dynamically resizes, pads, normalizes and tiles it
+    based on the image aspect ratio and the number of image tiles we allow.
+
+    The algorithm will NOT distort the image to fit a certain aspect ratio, because
+    that leads to a significant degradation in image quality.
+
+    The user can choose if they want to allow upscaling by using the flag resize_to_max_canvas.
+
+    For example, if an input image is of size 300x800, and we want to allow
+    a maximum of 16 image tiles, with side 224px, then:
+
+    If resize_to_max_canvas=False, then:
+    best_resolution = (448, 896) # smallest canvas, up to 16 tiles, that doesn't require downscaling
+    image is NOT resized
+    image is padded (300, 800) -> 448,896
+    Image is tiled 2x4, for a final output shape of (8, 3, 224, 224)
+
+    If resize_to_max_canvas=True, then:
+    best_resolution = (448, 1344) # canvas that allows maximum upscaling, with minimum padding, up to 16 tiles
+    image is resized without distortion (300,800) -> (448, 1194) #448 is the limiting side for the resize
+    image is padded (448, 1194) -> (448, 1344)
+    Image is tiled 2x5, for a final output shape of (10, 3, 224, 224)
+
+    Args:
+        image_mean (Optional[List[float]]): Mean values of each channel, used for normalization.
+            Should be the same used for the pre-trained model. If None, no normalization is performed.
+        image_std Union[float, List[float]]]): Standard deviation values of each channel, used for normalization.
+            Should be the same used for the pre-trained model. If None, no normalization is performed.
+        possible_resolutions (Optional[List[Tuple[int, int]]]): List of possible resolutions as tuples (height, width).
```

Comment:
maybe I'm still a bit noob but it's not clear to me what possible_resolutions mean, could you add another sentence describing what possible resolutions means in this case? and why multiple?